homeassistant/components/twinkly/light.py

```diff
@@ -321,14 +331,55 @@ async def async_update(self) -> None:
 
     async def async_update_movies(self) -> None:
         """Update the list of movies (effects)."""
-        movies = await self._client.get_saved_movies()
-        _LOGGER.debug("Movies: %s", movies)
-        if movies and "movies" in movies:
-            self._movies = movies["movies"]
+        try:
+            movies = self._valid_twinkly_response(
+                await self._client.get_saved_movies(), check_for="movies"
+            )
+        except TwinklyError as error:
+            _LOGGER.warning(error)
+            return
+        self._movies = movies["movies"]
 
     async def async_update_current_movie(self) -> None:
         """Update the current active movie."""
-        current_movie = await self._client.get_current_movie()
-        _LOGGER.debug("Current movie: %s", current_movie)
-        if current_movie and "id" in current_movie:
-            self._current_movie = current_movie
+        try:
+            self._current_movie = self._valid_twinkly_response(
+                await self._client.get_current_movie()
+            )
+        except TwinklyError as error:
+            _LOGGER.warning(error)
+
+    async def async_update_current_color(self) -> None:
+        """Update the current active color."""
+        try:
+            current_color = self._valid_twinkly_response(
+                await self._client.get_current_colour()
+            )
+        except TwinklyError as error:
+            _LOGGER.warning(error)
+            return
```

Comment:
This pattern is now repeated multiple times, maybe use a decorator instead?

Reply:
Added a decorator and a wrapper class to simplify these.  Hope this is what you were looking for.